syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.cluster";
option java_outer_classname = "Cluster";
option objc_class_prefix = "CTR";

package cluster;

service ControlService {
  rpc StartTask (RunMessage) returns (stream PointBatch) {}
  rpc SuspendTask (Empty) returns (PointBatch) {}
  rpc ReadCurrentPoints (Empty) returns (PointBatch) {}
  rpc ResumeTask (Empty) returns (Empty) {}
  rpc TerminateTask (Empty) returns (PointBatch) {}
}

message Empty{}

message RunMessage {
  string function_name = 1;
  repeated double param = 2;
}

message Point {
  double x = 1;
  double y = 2;
  double z = 3;
}

message PointBatch {
  repeated Point point = 1;
}
